### ISSUE TITLE
Deploy appimage integration files

### ISSUE THREAD

**Issue by azubieta** [nomad-desktop/nx-software-center#13]
the software center must deploy the appimage desktop and icon files into the system instead of the appimaged

**Comment by probonopd** [comment]
> the software center must deploy the appimage desktop and icon files into the system instead of the appimaged

Why?

**Comment by star-buck** [comment]
I noted that with appimaged installed and running in Rolling, currently I
get only white icon placeholders in menu for new appimages until i logout
login again?

**Comment by probonopd** [comment]
Yes, but I don't think this can be solved by the software center deploying the appimage desktop and icon files into the system instead of `appimaged`. Because if we find out how to fix it, we can fix it in `appimaged` just as well.

I think this has to do with how the desktop environment processes icon directories. It seems like if a certain directory or subdirectory within `$HOME/.local/share/icons` was not yet there at the time when the desktop environment was started, then the icons put there are ignored until the user logs out and logs in again, thus restarting the desktop environment.

Is there a possibility to make the desktop environment aware "after the fact" about new directories added there?

**Comment by star-buck** [comment]
Thats a good tip, maybe simply adding empty directories of those by default
would help?

**Comment by probonopd** [comment]
> Thats a good tip, maybe simply adding empty directories of those by default would help?

Yes, it might. But it's not the cleanest solution, especially since we don't know in which paths some AppImages will put files. E.g., `share/icons/hicolor` vs. `share/icons/default` and the different sizes. I think the better solution is to find out how to tell the desktop manager properly how to look for those icons there without needing a restart.

**Comment by star-buck** [comment]
Any idea for a solution (for Plasma)?

**Comment by probonopd** [comment]
Unfortunately no, it would be best to ask someone who knows about icon handling in Plasma.

**Comment by probonopd** [comment]
Does running `xdg-icon-resource forceupdate` make a difference?

**Comment by azubieta** [comment]
I think that the solution for kde base desktops will be running
kbuildsycoca5.

**Comment by probonopd** [comment]
Does `xdg-icon-resource forceupdate` work as well? (Maybe it calls `kbuildsycoca5` internally?). If it doesn't, can you confirm that `kbuildsycoca5` does?

If someone can confirm, we should make `appimaged` run these.

**Comment by TheAssassin** [comment]
> Does xdg-icon-resource forceupdate work as well? 

No, doesn't have any effect. Neither does `kbuildsycoca5`.

**Comment by azubieta** [comment]
In order to remove the run-time dependency on appimaged (or any other 
I'm going to call "appimage_is_registered_in_system" and "appimage_unregister_in_system" right from the software center.

**Comment by TheAssassin** [comment]
@azubieta please make sure to check out `appimage_shall_not_be_registered` as well.
